// @lemoncode/fonk-min-number-validator/dist/@lemoncode/fonk-min-number-validator.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@lemoncode/fonk')) :
  typeof define === 'function' && define.amd ? define(['exports', '@lemoncode/fonk'], factory) :
  (global = global || self, factory(global['@lemoncode/fonk-min-number-validator'] = {}, global.Fonk));
}(this, function (exports, fonk) { 'use strict';

  var VALIDATOR_TYPE = 'MIN_NUMBER';
  var defaultMessage = 'The value must be greater than or equal to {{minValue}}';
  var setErrorMessage = function setErrorMessage(message) {
    return defaultMessage = message;
  };
  var defaultCustomArgs = {
    strictTypes: false,
    minValue: 0,
    inclusive: true
  };
  var setCustomArgs = function setCustomArgs(customArgs) {
    return defaultCustomArgs = Object.assign(Object.assign({}, defaultCustomArgs), customArgs);
  };

  var validateType = function validateType(value, args) {
    return !args.strictTypes || typeof value === 'number';
  };

  var validate = function validate(value, args) {
    return !isNaN(Number(value)) ? args.inclusive ? value >= args.minValue : value > args.minValue : false;
  };

  var isDefined = function isDefined(value) {
    return value !== void 0 && value !== null && value !== '';
  };

  var validator = function validator(fieldValidatorArgs) {
    var value = fieldValidatorArgs.value,
        _fieldValidatorArgs$m = fieldValidatorArgs.message,
        message = _fieldValidatorArgs$m === void 0 ? defaultMessage : _fieldValidatorArgs$m,
        _fieldValidatorArgs$c = fieldValidatorArgs.customArgs,
        customArgs = _fieldValidatorArgs$c === void 0 ? defaultCustomArgs : _fieldValidatorArgs$c;
    var args = Object.assign(Object.assign({}, defaultCustomArgs), customArgs);
    var succeeded = !isDefined(value) || validateType(value, args) && validate(value, args);
    return {
      succeeded: succeeded,
      message: succeeded ? '' : fonk.parseMessageWithCustomArgs(message, args),
      type: VALIDATOR_TYPE
    };
  };

  var validator$1 = /*#__PURE__*/Object.freeze({
    __proto__: null,
    setErrorMessage: setErrorMessage,
    setCustomArgs: setCustomArgs,
    validator: validator
  });

  exports.minNumber = validator$1;

  Object.defineProperty(exports, '__esModule', { value: true });

}));
